### ISSUE TITLE
Cmocka installation problem while following the presets tutorial

### ISSUE THREAD

**Issue by marcoCalipari** [nqminds/brski#7]
## Problem ##
Unable to complete the installation process following the steps provided with the cmake presets. Most of the problems I had revolved around cmocka.
At first I tried to build the stable 0.20 release. However I was not able to build it.
Then I cloned the repository as per its latest update, and I was able to build. However, I keep stumbling accross cmocka errors when running `make --build --preset linux -j4`
I followed the steps illustrated form the repo.

Thanks in advance for your time and consideration

## LOG ##

```
[ 62%] Building C object _deps/cmocka-build/src/CMakeFiles/cmocka.dir/cmocka.c.o

/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c: In function ‘_assert_ptr_equal’:

/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2116:10: error: ‘a’ may be used uninitialized [-Werror=maybe-uninitialized]

 2116 |     if (!ptr_values_equal_display_error(a, b)) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2111:36: note: accessing argument 1 of a function declared with attribute ‘access (none, 1)’
 2111 | void _assert_ptr_equal(const void *a,
      |                        ~~~~~~~~~~~~^

/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2116:10: error: ‘b’ may be used uninitialized [-Werror=maybe-uninitialized]
 2116 |     if (!ptr_values_equal_display_error(a, b)) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2112:36: note: accessing argument 2 of a function declared with attribute ‘access (none, 2)’
 2112 |                        const void *b,
      |                        ~~~~~~~~~~~~^
/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c: In function ‘_assert_ptr_not_equal’:
/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2126:10: error: ‘a’ may be used uninitialized [-Werror=maybe-uninitialized]
 2126 |     if (!ptr_values_not_equal_display_error(a, b)) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2121:40: note: accessing argument 1 of a function declared with attribute ‘access (none, 1)’
 2121 | void _assert_ptr_not_equal(const void *a,
      |                            ~~~~~~~~~~~~^
/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2126:10: error: ‘b’ may be used uninitialized [-Werror=maybe-uninitialized]
 2126 |     if (!ptr_values_not_equal_display_error(a, b)) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/marco/Desktop/brski/build/linux/_deps/cmocka-src/src/cmocka.c:2122:40: note: accessing argument 2 of a function declared with attribute ‘access (none, 2)’
 2122 |                            const void *b,
      |                            ~~~~~~~~~~~~^
cc1: some warnings being treated as errors
```

## System ##
Kali linux distro

**Comment by mcr** [comment]
Maybe a newer compiler than Alex is using.
These all seem like -Werror results.

**Comment by marcoCalipari** [comment]
Thanks a lot for your availability.
I can wait no problem, i'll wait for updates.

Thanks again

**Comment by marcoCalipari** [comment]
> Thanks a lot for your availability. I can wait no problem, i'll wait for updates.
> 
> Thanks again

I looked in the repo, and the only istance of `-Werror` is in the voucher CMakeList file: 
`add_compile_options($<$<CONFIG:Debug>:-Werror>)`
eliminating that one out, nothing changed, same error as before.

![image](https://github.com/nqminds/brski/assets/48682055/87e3144b-61a2-44b3-b9c3-9731fa42f62f)

**Comment by aloisklink** [comment]
PR https://github.com/nqminds/brski/pull/10 fixes this for me, (at least for GCC 12.1.0).

I've also made a merge request in CMocka to fix this upstream, see https://gitlab.com/cmocka/cmocka/-/merge_requests/66

@marcoCalipari, let me know if PR #10 fixes this problem for you. If it doesn't, let us know which compiler version you're using!

It should be printed when you first run `cmake`, e.g. like:

```console
user@pc:~/brski (main)$ CC=gcc-12 cmake --preset linux
Preset CMake variables:

  BUILD_OPENSSL3_LIB:BOOL="FALSE"
  CMAKE_BUILD_TYPE:STRING="Debug"
  EP_DOWNLOAD_DIR:PATH="/home/user/brski/build/dl"
  SANITIZE_ADDRESS:BOOL="FALSE"
  USE_CPPHTTPLIB_LIB:BOOL="TRUE"
  USE_VOUCHER_OPENSSL:BOOL="TRUE"
  USE_VOUCHER_WOLFSSL:BOOL="FALSE"

-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
# ...
```

---

> I looked in the repo, and the only istance of `-Werror` is in the voucher CMakeList file:

Ah, that's because the `-Werror` was in [CMocka's CMakeLists.txt file](https://gitlab.com/cmocka/cmocka/-/blob/master/CompilerChecks.cmake#L10)! Thanks for checking anyway.

**Comment by marcoCalipari** [comment]
Hello,

Thanks  a lot for you changes, now it builds without problems.
I will take care of closing the issue.

Fyi, although pointless my gcc compiler is:

`
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
`
